great_expectations/execution_engine/execution_engine.py

```diff
@@ -220,20 +220,24 @@ def resolve_metrics(
         if metrics is None:
             metrics = {}
 
-        resolved_metrics = {}
+        resolved_metrics: Dict[Tuple, MetricConfiguration] = {}
 
         metric_fn_bundle = []
         for metric_to_resolve in metrics_to_resolve:
+            metric_dependencies = {}
+            for k, v in metric_to_resolve.metric_dependencies.items():
+                if v.id in metrics:
+                    metric_dependencies[k] = metrics[v.id]
+                elif self._caching and v.id in self._metric_cache:
+                    metric_dependencies[k] = self._metric_cache[v.id]
+                else:
+                    raise ge_exceptions.MetricError(
```

Comment:
I'm not sure if I'm understanding this right, but if the metric has not been calculated yet (but may be later) would this raise a MetricError?

Reply:
@anthonyburdi This code focuses on evaluating the given metric.  For it, all of its dependencies *must* already be calculated and stored/cached.  Hence, the code reflects this logic.